### PR TITLE
Creating database for the server using postgre and libpq++

### PR DESCRIPTION
**Pull request by dxlpi**
I'm not done with creating the database! I just wanted to get checked by you guys to see if I'm on the right track or not. Any suggestions or comments are welcomed

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,8 +5,10 @@ project(alicia-server
 option(BUILD_TESTS "Build tests" ON)
 
 find_package(Boost REQUIRED)
+find_package(PkgConfig REQUIRED)
```
Comment (rgnter) [inline CMakeLists.txt:8]: This will not work on Windows.

File: CMakeLists.txt [hunk 2]
```diff
@@ -5,8 +5,10 @@ project(alicia-server
...
+pkg_check_modules(LIBPQXX REQUIRED libpqxx)
 
-add_subdirectory(3rd-party)
+add_subdirectory(3rd-party/spdlog)
```
Comment (rgnter) [inline CMakeLists.txt:11]: This was fine like it was before. 
`3rd-party` folder contains CMakeLists.txt file which points to subdirectories.

File: src/.DS_Store
Comment (rgnter) [inline src/.DS_Store:1]: Add to .gitignore

**Review by rgnter — CHANGES_REQUESTED** [review]

**Comment by rgnter** [comment]
Instead of using pkg-config, do the following:

1. Clone a submodule of libpqxx:
```
git submodule add https://github.com/jtv/libpqxx.git 3rd-party/libpqxx
```

2. Add the submodule to the CMakeLists of the `3rd-party` folder.
```diff
add_subdirectory(spdlog)
+add_subdirectory(libpqxx)
```

3. Link the target provided by libpqxx with our poject.
```cmake
target_link_libraries(alicia-server
        PRIVATE project-properties alicia-libserver spdlog::spdlog pqxx)
```

**Comment by dxlpi** [comment]
@rgnter I made the changes accordingly! Please review the changes and lmk if there's anything else I need to fix. Thanks :)

File: CMakeLists.txt [hunk 3]
```diff
@@ -5,6 +5,7 @@ project(alicia-server
 option(BUILD_TESTS "Build tests" ON)
 
 find_package(Boost REQUIRED)
+pkg_check_modules(LIBPQXX REQUIRED libpqxx)
```
Comment (rgnter) [inline CMakeLists.txt:8]: You don't need this anymore. If you do, the build isn't setup correctly.

**Review by rgnter — CHANGES_REQUESTED** [review]

**Comment by rgnter** [comment]
For future reference, I recommend rebasing your squashed feature branch instead of merging it with master. This way the commits that don't belong to this MR will not show up.

**Pull request closed without merging**